tests/unit/customizations/history/test_db.py

```diff
@@ -0,0 +1,711 @@
+        self.assertEqual(request_id, '')
+
+    def test_commit_count_matches_write_count(self):
+        records_to_write = 10
+        for _ in range(records_to_write):
+            self._write_record({
+                'event_type': 'foo',
+                'payload': '',
+                'source': 'TEST',
+                'timestamp': 1234
+            })
+        self.assertEqual(self.fake_connection.commits, records_to_write)
+
+    def test_can_write_cli_version_record(self):
+        insert_stmt, written_record = self._write_record({
+            'event_type': 'CLI_VERSION',
+            'payload': ('aws-cli/1.11.184 Python/3.6.2 Darwin/15.6.0 '
+                        'botocore/1.7.42'),
+            'source': 'TEST',
+            'timestamp': 1234
+        })
+
+        self.assert_insert_statement_structure_correct(insert_stmt)
+        self.assertEqual(
+            written_record[2:],
+            ('TEST', 'CLI_VERSION', 1234,
+             '"aws-cli/1.11.184 Python/3.6.2 Darwin/15.6.0 botocore/1.7.42"')
+        )
+        identifier, request_id = written_record[:2]
+        self.assertTrue(self.UUID_PATTERN.match(identifier))
+        self.assertEqual(request_id, '')
+
+    def test_can_write_cli_arguments_record(self):
+        insert_stmt, written_record = self._write_record({
+            'event_type': 'CLI_ARGUMENTS',
+            'payload': ['s3', 'ls'],
+            'source': 'TEST',
+            'timestamp': 1234
+        })
+
+        self.assert_insert_statement_structure_correct(insert_stmt)
+        self.assertEqual(
+            written_record[2:],
+            ('TEST', 'CLI_ARGUMENTS', 1234, '["s3", "ls"]')
+        )
+        identifier, request_id = written_record[:2]
+        self.assertTrue(self.UUID_PATTERN.match(identifier))
+        self.assertEqual(request_id, '')
+
+    def test_can_write_api_call_record(self):
+        insert_stmt, written_record = self._write_record({
+            'event_type': 'API_CALL',
+            'payload': {
+                'service': 's3',
+                'operation': 'ListBuckets',
+                'params': {},
+            },
+            'source': 'TEST',
+            'timestamp': 1234
+        })
+
+        self.assert_insert_statement_structure_correct(insert_stmt)
+        self.assertEqual(
+            written_record[2:],
+            ('TEST', 'API_CALL', 1234, json.dumps({
+                'service': 's3',
+                'operation': 'ListBuckets',
+                'params': {},
+            }))
+        )
+        identifier, request_id = written_record[:2]
+        self.assertTrue(self.UUID_PATTERN.match(identifier))
+        self.assertTrue(self.UUID_PATTERN.match(request_id))
+
+    def test_can_write_http_request_record(self):
+        insert_stmt, written_record = self._write_record({
+            'event_type': 'HTTP_REQUEST',
+            'payload': {
+                'method': 'GET',
+                'headers': {},
```

Comment:
Should this use the `CaseInsensitiveDict` class?

Reply:
yes